app/models/tenant_quota.rb

```diff
@@ -1,3 +1,72 @@
 class TenantQuota < ActiveRecord::Base
   belongs_to :tenant
+
+  QUOTAS = {
+    :cpu_allocated => {
+      :unit          => :mhz,
+      :format        => :mhz,
+      :text_modifier => "Mhz"
+    },
+    :mem_allocated => {
+      :unit          => :bytes,
+      :format        => :gigabytes_human,
+      :text_modifier => "GB"
+    },
+    :storage_allocated => {
+      :unit          => :bytes,
+      :format        => :gigabytes_human,
+      :text_modifier => "GB"
+    },
+    :vms_allocated => {
+      :unit          => :fixnum,
+      :format        => :general_number_precision_0,
+      :text_modifier => "Count"
+    },
+    :templates_allocated => {
+      :unit          => :fixnum,
+      :format        => :general_number_precision_0,
+      :text_modifier => "Count"
+    }
+  }
+
+  NAMES = QUOTAS.stringify_keys
+
+  validates :name, :inclusion => {:in => NAMES}
+  validates :unit, :value, :presence => true
+
+  def self.available
+    return @available if @available
+
+    @available = QUOTAS.each_with_object({}) do |q, h|
```

Comment:
Is this because we don't have a QuotaDefinition model that would be be seeded by QUOTAS?  It feels like we could simplify this loop with `ui_description` method on QuotaDefinition.  I guess my concern is that it feels like this method should be returning QuotaDefinition objects, not hashes.  

Feel free to rename QuotaDefinition to something better in your head.  Naming is hard.